### ISSUE TITLE
Accessibility evaluation

### ISSUE THREAD

**Issue by agbeltran** [ral-facilities/datagateway#738]
**Description**:
We want to make sure that the web app is accesible. 

Some resources to consider:
https://www.gov.uk/government/publications/doing-a-basic-accessibility-check-if-you-cant-do-a-detailed-one

https://www.gov.uk/guidance/make-your-website-or-app-accessible-and-publish-an-accessibility-statement?utm_source=CampaignPage&utm_campaign=access_regs#decide-how-to-check-the-accessibility-problems-on-your-website-or-mobile-app

https://wave.webaim.org/

**Acceptance criteria**:
- [ ] evaluation results

**Comment by agbeltran** [comment]
https://www.w3.org/TR/WCAG21/

**Comment by louise-davies** [comment]
@MRichards99 this should probably now be handed over to someone else, as you're now busy with datagateway-api issues

**Comment by MRichards99** [comment]
@louise-davies who would you like this to be handed over to? I'm happy to write up a few notes and meet with the newly assigned person to go through where I got to and what needs to be done once this is decided.

**Comment by louise-davies** [comment]
we were just given feedback that the blues & purples we use aren't very visible for a colour blind user in darkmode. We may have to come back to this user for more specific feedback, but we should also be considering this. For now I assume that they'd be fine with using light mode...

**Comment by joshuadkitenge** [comment]
I have found a tool that can simulate different types of colour blindness, and I'm going to redo my contrast evaluation using that.

**Comment by joelvdavies** [comment]
@louise-davies Using the tool @joshuadkitenge suggested I can immediately see the buttons may cause a problem 
![image](https://user-images.githubusercontent.com/90245114/141495888-5cf8938a-e6b6-474b-9864-4b95cddaeb1d.png). We could make the buttons lighter, or as Joshua pointed out we could create a high contrast dark mode theme like Github does.

**Comment by louise-davies** [comment]
@joelvdavies I think making a separate theme would probably be suitable as otherwise we'd be chasing the probably impossible dream of getting one theme looking nice for both colourblind and non-colourblind people (although GitHub's colourblind themes are still in beta so we shouldn't beat ourselves up too much about this)

**Comment by joshuadkitenge** [comment]
## WCAG 1.4.10 (AA): Reflow
-  "Content can be presented without loss of information or functionality, and without requiring scrolling in two dimensions" <br><br>
    - Table view doesn't show any data above 250% zoom. (400% required) <br><br>

## WCAG 2.1.1 (A): Keyboard , WCAG 2.1.3 (AAA): Keyboard (No Exceptions)
-  "All functionality of the content is operable through a keyboard interface without requiring specific timings for individual keystrokes, except where the underlying function requires input that depends on the path of the user's movement and not just the endpoints." <br><br>
    - the Only reason for failure is that columns cannot be resized in table view using keyboard-only navigation.<br><br>

## WCAG 2.2.6 (AAA): Timeouts
-  "Users are warned of the duration of any user inactivity that could cause data loss, unless the data is preserved for more than 20 hours when the user does not take any actions." <br><br>
    - If cart is lost on anon user, maybe we should warn when their session is nearing expiry? <br><br> 

## WCAG 2.4.9 (AAA): Link Purpose (Link Only)
-  "A mechanism is available to allow the purpose of each link to be identified from link text alone, except where the purpose of the link would be ambiguous to users in general." <br><br>
    - The lack of context would make link text (on table and card view on investigation titles) unclear <br><br> 

## WCAG 2.5.5 (AAA): Target Size
-  "The size of the target for pointer inputs is at least 44 by 44 CSS pixels" <br><br>
    - Checkboxes on table view are only 20x20 and drop down is only 24x24 <br><br> 

## WCAG 3.1.4, 3.1.5, 3.16 (AAA): Abbreviations, Reading Level and Pronunciation
-  There's a lot of complex text data stored in ICAT<br><br>

## WCAG 3.1.4, 3.1.5, 3.16 (AAA): Abbreviations, Reading Level and Pronunciation
-  There's a lot of complex text data stored in ICAT<br><br>

## WCAG 3.3.5 (AAA): Help 
- "Context-sensitive help is available."
      - There is help text in the form of the tour, but that's more general rather than "related to the function currently being performed" <br><br>

## WCAG 3.3.6 (AAA): Error Prevention (All) 
- For Web pages that require the user to submit information, at least one of the following is true:
  - Reversible 
        - Submissions are reversible.
  - Checked 
        -  Data entered by the user is checked for input errors and the user is provided an opportunity to correct them. 
  - Confirmed 
            - A mechanism is available for reviewing, confirming, and correcting information before finalizing the submission.  <br><br>
        
- For this to pass, we should add a confirmation on download (when a user inputs email address etc.) although I wonder if this would be annoying for a user

## WCAG 4.1.3 (AA): Status Messages
-  "In content implemented using markup languages, status messages can be programmatically determined through role or properties such that they can be presented to the user by assistive technologies without receiving focus." <br><br>
    - Check whether the warnings and notifications have a relevant property or aria label <br><br> 

# More information 
- https://stfc365.sharepoint.com/:x:/s/TopcatRedevelopment/EdclB82juphJo1Qd1NYN04YBzqRcFn9kOiHcTai_WiyJ9Q?email=joshua.kitenge%40stfc.ac.uk&e=ulKIPG
- https://github.com/ral-facilities/scigateway/wiki/DataGateway-Accessability

**Comment by joshuadkitenge** [comment]
- Adding a minimum height to table view (everywhere) 

- https://github.com/ral-facilities/datagateway/pull/1063/files (searchPageContainer , line 154)

- This would allow WCAG 2.1 AA Reflow to pass
